scipy/linalg/tests/test_lapack.py

```diff
@@ -1654,3 +1654,158 @@ def test_getc2_gesc2():
         else:
             assert_array_almost_equal(desired_cplx.astype(dtype),
                                       x/scale, decimal=4)
+
+
+def generate_random_dtype_array(shape, dtype):
+    # generates a random matrix of desired data type of shape
+    if dtype in COMPLEX_DTYPES:
+        return (np.random.rand(*shape)
+                + np.random.rand(*shape)*1.0j).astype(dtype)
+    return np.random.rand(*shape).astype(dtype)
+
+
+@pytest.mark.parametrize("size", [(6, 5), (5, 5)])
+@pytest.mark.parametrize('dtype', DTYPES)
+@pytest.mark.parametrize('gejsv_lambda',
+                         [(lambda A, gejsv: gejsv(A)),  # base default case
+                          (lambda A, gejsv: gejsv(A, joba="E")),  # joba
+                          (lambda A, gejsv: gejsv(A, joba="F")),
+                          (lambda A, gejsv: gejsv(A, joba="G")),
+                          (lambda A, gejsv: gejsv(A, joba="A")),
+                          (lambda A, gejsv: gejsv(A, joba="R")),
+                          (lambda A, gejsv: gejsv(A, jobu="F")),  # jobu
+                          (lambda A, gejsv: gejsv(A, jobu="W")),
+                          (lambda A, gejsv: gejsv(A, jobv="J")),  # jobv
```

Comment:
This is not all the options we need tests for. I might not have addressed them all in the original issue, so let's talk about each possibility.